File: rust_exercises/leetcode/Problem724/Solution.java
public class Solution {
    public int pivotIndex(int[] nums) {
        int leftSum = 0;
        int rightSum = 0;
        for (int i = 0; i < nums.length; i += 1) {
            rightSum += nums[i];
        }
        
        int pivot = 0;
        while (pivot < nums.length) {
            rightSum -= nums[pivot];
            if (leftSum == rightSum) {
                return pivot;
            }
            leftSum += nums[pivot];
            pivot += 1;
        }
        
        return -1;
    }
}

File: rust_exercises/leetcode/Problem724/solution.rs
impl Solution {
    pub fn pivot_index(nums: Vec<i32>) -> i32 {
        let mut left_sum: i32 = 0;
        let mut right_sum: i32 = nums.iter().sum();
        
        let mut pivot: usize = 0;
        
        while pivot < nums.len() {
            right_sum -= nums[pivot];
            if left_sum == right_sum {
                return pivot as i32;
            }
            left_sum += nums[pivot];
            pivot += 1;
        }
        
        -1
    }
}
